modules/ducktests/tests/ignitetest/tests/snapshot_test.py

```diff
@@ -0,0 +1,162 @@
+from ignitetest.services.utils.ignite_configuration.data_storage import DataRegionConfiguration
+from ignitetest.utils import ignite_versions
+from ignitetest.utils.ignite_test import IgniteTest
+from ignitetest.utils.version import DEV_BRANCH, IgniteVersion
+
+
+# pylint: disable=W0223
+class SnapshotTest(IgniteTest):
+    """
+    Test Snapshot.
+    """
+    NUM_NODES = 4
+
+    SNAPSHOT_NAME = "test_snap"
+
+    @cluster(num_nodes=NUM_NODES)
+    @ignite_versions(str(DEV_BRANCH))
+    def snapshot_test(self, ignite_version):
+        """
+        Test Snapshot.
+        1. Start of ignite cluster.
+        2. Activate cluster.
+        3. Load.
+        4. Idle verify dump1.
+        5. Snapshot.
+        6. Load.
+        7. Idle verify dump2.
+        8. Сhecking that dump1 and dump2 are differentю
+        9. Stop ignite and replace db from snapshot
+        10. Run ignite cluster.
+        11. Idle verify dump3.
+        12. Checking the equality of dump1 and dump3.
+        """
+        data_cfg = DataStorageConfiguration(default=DataRegionConfiguration(persistent=True))
+
+        ignite_config = IgniteConfiguration(
+            version=IgniteVersion(ignite_version),
+            data_storage=data_cfg,
+        )
+
+        service = IgniteService(self.test_context, ignite_config, num_nodes=self.NUM_NODES - 1)
+        service.start()
+
+        control_utility = ControlUtility(service, self.test_context)
+        control_utility.activate()
+
+        client_config = ignite_config._replace(client_mode=True, data_storage=None)
+
+        streamer = IgniteApplicationService(
+            self.test_context,
+            client_config,
+            java_class_name="org.apache.ignite.internal.ducktest.tests.UuidStreamerApplication",
+            params={"cacheName": "test-cache"})
+
+        load(streamer)
+
+        node = service.nodes[0]
+
+        check_idle_verify(control_utility)
+        check_validate_indexes(control_utility)
+        dump_1 = get_dump_path(control_utility, node)
+
+        self.logger.warn("Path to dump_1=" + dump_1)
+
+        data = control_utility.snapshot_create(self.SNAPSHOT_NAME)
+
+        self.logger.warn(data)
+
+        load(streamer)
+
+        dump_2 = get_dump_path(control_utility, node)
+
+        self.logger.warn("Path to dump_2=" + dump_2)
+
+        diff = node.account.ssh_output(f'diff {dump_1} {dump_2}', allow_fail=True)
+        assert len(diff) != 0, diff
+
+        service.rename_db(new_db_name='old_db')
+        service.copy_snap_to_db(self.SNAPSHOT_NAME)
+
+        service.run()
+
+        control_utility.activate()
+
+        check_idle_verify(control_utility)
+        check_validate_indexes(control_utility)
+        dump_3 = get_dump_path(control_utility, node)
+
+        self.logger.warn("Path to dump_3=" + dump_2)
+
+        diff = node.account.ssh_output(f'diff {dump_1} {dump_3}', allow_fail=True)
+        assert len(diff) == 0, diff
```

Comment:
Am I correct that equality of dump1 and dump3 is achieved by the fact that streamer loads data the same time (1 min long for every load)?

Reply:
No, this is because snapshot is restored

Reply:
dump3 is equal to dump1
because we restored from a snapshot and the data in the cluster is the identical
idle_verify --dump
generates a file that contains itself counters and data hash sums

Partition: PartitionKeyV2 [grpId=623628935, grpName=test-cache, partId=0]
Partition instances: [PartitionHashRecordV2 [isPrimary=true, consistentId=ducker03, updateCntr=73, partitionState=OWNING, size=73, partHash=-878633494]]

Reply:
Ah, ok, I see.